docs/tools/testing/icetea.md

```diff
@@ -0,0 +1,192 @@
+    - To enumerate connected boards.
+  - [mbed-flasher](https://github.com/ARMmbed/mbed-flasher).
+    - To flash devices automatically.
+  - Icetea installation installs both of these automatically.
+  
+**OS specific:**
+
+Linux:
+
+- `python-dev` and `python-lxml`
+`sudo apt-get install python-dev python-lxml`
+- For documentation, install sphinx:
+`apt-get install python-sphinx`
+
+OS X:
+
+- [XCode developer tools](http://osxdaily.com/2014/02/12/install-command-line-tools-mac-os-x/).
+- [lxml](http://lxml.de/installation.html#installation):
+`STATIC_DEPS=true sudo pip install lxml`
+- For documentation, install sphinx:
+`sudo port install py27-sphinx`, `sudo port select --set python python27`, `sudo port select --set sphinx py27-sphinx`
+
+Windows:
+
+- `python-lxml` installation usually requires build tools. You can, however, install it from prebuilt binaries.
+    - Download the binary for your system from the internet.
+    - Navigate to the directory where you downloaded the binary, and install it with `pip install <insert_file_name>`.
+- For documentation, install sphinx with pip:
+`pip install sphinx`
+
+#### Optional dependencies
+
+- To decorate your console log with colors, install the coloredlogs module using pip:
+`pip install coloredlogs`
+    - Windows does not support the coloredlogs module.
+- [valgrind](http://valgrind.org).
+- [gdb](https://www.gnu.org/software/gdb/).
+
+#### Installation instructions
+
+```
+git clone https://github.com/ARMmbed/icetea.git
+cd icetea
+python setup.py install
+```
+
+**Linux**
+
+To run test cases with hardware in Linux without sudo rights:
+
+```
+sudo usermod -a -G dialout username
+Log out & log in back to Linux
+```
+
+This command adds the user `username` to the `dialout` group and grants the required permissions to the USB ports.
+
+### Use
+
+To print the help page:
+
+`icetea --help`
+
+All CLI parameters are described in [icetea.md](https://github.com/ARMmbed/icetea/blob/master/doc/Icetea.md)
+
+To list all local testcases from the `./testcases` subfolder:
+
+`icetea --list`
+
+#### Test case API
+
+To execute a single command line:
+
+` execute_command(<dut>, <cmd>, (<arguments>) ) `
+
+or alias `command(...)`
+
+
+```
+dut[number]             #dut index (1..)
+dut[string]             #dut nick name, or '*' to execute command in all duts
+cmd[string]             #command to be executed
+arguments[dictionary]   #optional argument list
+        wait = <boolean>           # whether retcode is expected before continue next command. True (default) or False
+        expected_retcode = <int>    # expected return code (default=0)
+        timeout=<int>              # timeout, if no retcode receive
+```
+
+### Documentation
+
+You can build HTML documentation for Icetea using sphinx. The source for the documentation is located in [doc-source](https://github.com/ARMmbed/icetea/tree/master/doc-source). For installation of sphinx, please see the [installation instructions](#installation).
```

Comment:
Is this right? None of our other tools build documentation like this.

Reply:
I just don't understand why they need to build it. Why don't we publish it?

Reply:
This file is copied from icetea repository. There is separated issue about reviewing it IOTDOCS-459